boards/silabs/dev_kits/xg24_ek2703a/Kconfig.xg24_ek2703a

```diff
@@ -0,0 +1,5 @@
+# Copyright (c) 2021, Sateesh Kotapati
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_XG24_EK2703A
+	select SOC_PART_NUMBER_EFR32MG24B310F1536IM48
```

Comment:
The SoC OPN used on EK2703A is `EFR32MG24B210F1536IM48` (i.e. a 2 instead of a 3). It would need to be added to `dts/arm/silabs` and `soc/silabs`, as it isn't added in Zephyr yet. The difference between the two is that the 210 has 4 more GPIO pins, but does not have the high accuracy mode for IADC or the MVP accelerator. Neither of these are represented in device tree yet, so it makes no practical difference right now, but it would be good for the SoC selection to be correct in preparation for when they are.

Reply:
Will do that along with the cleanup mentioned below in a separate PR.